design/secretless-e2e-tests-dnd.md

```diff
@@ -0,0 +1,256 @@
+declare -A conjur_service_config
+conjur_service_config=(
+["service_name"]="conjur_abc",
+["config"]="conjur.yaml",
+["start"]="conjur_start.sh",
+["stop"]="conjur_stop.sh"
+)
+
+declare -A mysql_service_config
+mysql_service_config=(
+["service_name"]="mysql_abc",
+["config"]="mysql.yaml",
+["start"]="mysql_start.sh",
+["stop"]="mysql_stop.sh"
+)
+``` 
+1. There is a test case written in Gherkin for validating using the MySQL service connector with Conjur.
+```
+Feature: Guess the word
+  
+  @conjur @mysql @mysql-conjur
+  Scenario: Connection to MySQL
+    Given Secretless is running as a sidecar in Kubernetes
+    Then the app is able to connect to MYSQL on "localhost:332"
+
+```
+2. Cucumber will register that there's a test case that requires Conjur, MySQL and MySQL credentials in Conjur.
+3. The developer calls `cucumber` to start the testss
+4. As a result of the annotations Cucumber will first call the engine to create the services.
+5. The configuration values
+ 
+TODO: provide examples of terms above
+TODO: split out components of the architecture by owner (infra, project developer etc.)
+
+## Interfaces
+
+TBD
+
+### Opaque API for lifecycle management of dependency resources
+
+The API makes it possible to implement service lifecycle methods in whatever way makes sense. That could be literal bash, docker-compose, calling an HTTP API to grab the service from a pool etc. This makes it possible to have similar experience between different environments (Jenkins, developer machine etc.). The same exact commands would be called on Jenkins as on the developer machine. The action of the commands would depend on the environment. For example, the developer machine can use a less resource intensive implementation of the service and Jenkins can run something that more closely approximates customer environments.
+
+Steps towards implementation
+
+1. In the short term we can change `kubernetes-conjur-demo` so that the proposed interface masks calls to `kubernetes-conjur-deploy`
+2. Later the infrastructure team or whoever can modify `kubernetes-conjur-deploy` jumping into cleaning up and improving implementation details without adversely affecting existing tests.
+3. From then on efforts to optimise for startup time, performance, customer-environment accuracy can continue. Once again, without adversely affecting existing tests.
+
+## Testing
+
+TBD
+
+## Infrastructure
+
+The orchestration of dependencies (fixtures, services etc.) will need to be run somewhere. Some dependencies such as Conjur followers need to be run on the same infrastructure as the tests. Some dependencies such as target services can and perhaps should be run on separate infrastructure from the tests.
+
+Q: It's not clear if this requires new infrastructure
+
+## Security
+
+TBD
+
+This section is still an open question. It's not clear at the moment if there are any security implications to the work under consideration.
+
+## Documentation
+
```

Comment:
worth adding a note here that the documentation is contributor-facing, and there's no requirement for user-facing documentation